
class MyError {

  msg: string

}

function test2(content: string): Result<i32, MyError> {
  if content.length == 0 {
    const err: Result<i32, MyError> = Error(MyError{
      msg: "null string"
    });
    err
  } else {
    Ok(0)
  }
}

function test1(content: string): Result<string, MyError> {
  const str = "finished";
  const num = test2(content)?;
  print(num);
  Ok(str)
}

function main() {
  test1("some");
  const result2 = test1("");
  match result2 {
    case Error(myErr) => print(myErr.msg)
    case _ => print("no error")
  }
}
